### PR TITLE
Landscaper control-plane component part 2: add flow + certificate generation

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:

Next part of the landscaper control plane component.

Adds
- changes to the import API of the landscaper control plane. The main part here is that it is possible to provide optional secret references in addition to providing certificate values directly.
- the skeleton flow containing everything related to generating, validation, and export of certificates.
- updates to the README

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

Adds the type "secrets" to the etcd encryption configuration. The configuration for the Gardener API server looks like this:

 ```
 apiVersion: apiserver.config.k8s.io/v1
  kind: EncryptionConfiguration
  resources:
  - providers:
    - aescbc:
        keys:
        - ...
    - identity: {}
    resources:
    - secrets
    - controllerdeployments.core.gardener.cloud
    - controllerregistrations.core.gardener.cloud
    - shootstates.core.gardener.cloud
```

- Does not yet contain tests for the steps in the deployment flow. This will be provided in a later PR.

The flow is already runnable, though does not deploy anything to the Garden or runtime cluster yet.
However, generated certificates are exported to the local filesystem.

**Release note**:

```other operator

```

### CODE REVIEW

**Comment by danielfoehrKn** [comment]
@achimweigel already took a look at the PR, but there are too many Gardener specifics in there. 
It will probably make more sense to add him in a subsequent PR that introduces the landscaper specific files.

File: landscaper/common/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package utils
```
Comment (rfranzke) [inline landscaper/common/utils/utils.go:15]: license header missing

File: landscaper/pkg/controlplane/controller/flow_tls_serving_certificates.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	// certificate of the Gardener API server
+	caCert, err := secretsutil.LoadCertificate("", []byte(*ca.Key), []byte(*ca.Crt))
+	if err != nil {
+		return nil, err
+	}
+
+	date := time.Now().UTC().AddDate(10, 0, 0)
+	validity := date.Sub(time.Now().UTC())
+	certConfig := &secretsutil.CertificateSecretConfig{
+		// Name:       fmt.Sprintf("%s-tls", serviceName), // not used
```
Comment (rfranzke) [inline landscaper/pkg/controlplane/controller/flow_tls_serving_certificates.go]: ?

Comment (danielfoehrKn) [reply landscaper/pkg/controlplane/controller/flow_tls_serving_certificates.go]: Thanks for spotting, overlooked that.

**Comment by rfranzke** [comment]
/ping @danielfoehrKn  - can you please finalize this PR so that we can merge it and remove the 'draft' label from #4865?

**Review by rfranzke — APPROVED** [review]

**Pull request merged**